src/tools/PrizeTierController/EditPrizeTiersModal/EditPrizeTierHistoryForm.tsx

```diff
@@ -0,0 +1,363 @@
+import { ErrorMessage } from '@components/ErrorMessage'
+import { StyledInput } from '@components/Input'
+import { Label } from '@components/Label'
+import { Button, ButtonTheme, ButtonSize } from '@pooltogether/react-components'
+import { calculate, calculateNumberOfPrizesForTierIndex } from '@pooltogether/v4-utils-js'
+import { EditPrizeTierFormValues } from '@prizeTierController/interfaces'
+import { getLastNonZeroTierIndex } from '@prizeTierController/utils/getLastNonZeroTierIndex'
+import classNames from 'classnames'
+import { utils } from 'ethers'
+import { useEffect, useState } from 'react'
+import {
+  FieldErrorsImpl,
+  useForm,
+  UseFormRegister,
+  UseFormWatch,
+  useFieldArray,
+  Control
+} from 'react-hook-form'
+import useDebouncedCallback from 'beautiful-react-hooks/useDebouncedCallback'
+
+export const EditPrizeTierHistoryForm = (props: {
+  onSubmit: (prizeTier: EditPrizeTierFormValues) => void
+  defaultValues?: Partial<EditPrizeTierFormValues>
+  decimals: number
+  displayAdvancedOptions?: boolean
+}) => {
+  const { onSubmit, defaultValues, decimals, displayAdvancedOptions } = props
+  const {
+    handleSubmit,
+    register,
+    control,
+    watch,
+    setValue,
+    formState: { errors, isValid }
+  } = useForm<EditPrizeTierFormValues>({
+    mode: 'onChange',
+    defaultValues,
+    shouldUnregister: true
+  })
+
+  // Reactively calculating prize value based on bitrange and tier changes:
+  const bitRange = parseInt(watch('bitRangeSize'))
+  const tierValues = watch('tiers').map((item) => Number(item.value))
+  const updatePrizeValue = useDebouncedCallback((bitRange: number, tierValues: number[]) => {
+    if (!!bitRange && tierValues.length > 0 && tierValues.every((value) => !Number.isNaN(value))) {
+      const totalValueOfPrizes = tierValues.reduce(
+        (a, b, i) => a + b * calculateNumberOfPrizesForTierIndex(bitRange, i),
+        0
+      )
+      setValue('prize', totalValueOfPrizes.toString())
+    }
+  })
+  useEffect(() => {
+    updatePrizeValue(bitRange, tierValues)
+    return () => updatePrizeValue.cancel()
+  }, [bitRange, tierValues])
+
+  return (
+    <form onSubmit={handleSubmit((v) => onSubmit(v))} className='flex flex-col' autoComplete='off'>
+      <div className='flex flex-col'>
+        <BitRangeSize errors={errors} register={register} isHidden={!displayAdvancedOptions} />
```

Comment:
nit: Wrap these advanced options in a div that has a subtle background to group it all together visually so it's easier to understand.

Reply:
Fixed in 12a6b88 along with other improvements.